### PR TITLE
Image List Component

### PR DESCRIPTION
**Pull request by spbyrne**
This adds list support for the image field type and improves the UI for the base image field. 

Screenshot 2023-03-14 at 10 32 59 AM

closes ENG-845

### CODE REVIEW

File: packages/@tinacms/toolkit/src/packages/fields/components/ImageUpload/ImageUpload.tsx [hunk 1]
```diff
@@ -22,25 +23,29 @@ interface ImageUploadProps {
   loading?: boolean
 }
 
+const isStringImage = (src: string) => {
```
Comment (jamespohalloran) [inline packages/@tinacms/toolkit/src/packages/fields/components/ImageUpload/ImageUpload.tsx:26]: Can we use this existing function for this?
https://github.com/tinacms/tinacms/blob/a464f90e3e4e0bb6da0a716979807166a6f30a32/packages/@tinacms/toolkit/src/components/media/utils.ts#L22

@logan-anderson modified the Regex to work with Tina Cloud images, so not sure if that's something we'll need here or not

**Review by jamespohalloran — CHANGES_REQUESTED** [review]
UI looks great! one comment on the isMedia fn, which might lead to a bug on prod

File: packages/@tinacms/toolkit/src/packages/fields/components/ImageUpload/ImageUpload.tsx (hunk 1 above)
Comment (spbyrne) [reply packages/@tinacms/toolkit/src/packages/fields/components/ImageUpload/ImageUpload.tsx:26]: Ah! I see. This was mainly a quick stab at improving the `src && (src.includes('png') || src.includes('svg'))` logic.

**Review by jamespohalloran — APPROVED** [review]
Looks good!

**Pull request merged**